src/fds/tests/test_sg_list.cpp

```diff
@@ -22,81 +23,161 @@ SISL_OPTION_GROUP(test_sg_list,
                   (num_threads, "", "num_threads", "number of threads",
                    ::cxxopts::value< uint32_t >()->default_value("8"), "number"))
 
-struct SgListTest : public testing::Test {};
-
-// the iterator request size is same as iov size for each iov;
-TEST_F(SgListTest, TestIteratorAlignedSize) {
-
-    sisl::sg_iovs_t iovs;
-    iovs.push_back(iovec{nullptr, 1024});
-    iovs.push_back(iovec{nullptr, 512});
-    iovs.push_back(iovec{nullptr, 2048});
-    iovs.push_back(iovec{nullptr, 512});
-    uint32_t iov_size_total = 0;
-    for (const auto& v : iovs) {
-        iov_size_total += v.iov_len;
+static constexpr uint32_t SZ{sizeof(uint32_t)};
+
+class SgListTestBasic : public testing::Test {
+public:
+    void SetUp() override {
+        iovs.push_back(iovec{nullptr, 1024});
```

Comment:
I don't think it is a good idea to move the sequence here in setup. 
I see you are trying to re-use the same code, but the point of the test case is the sequence we are pushing back the iovs are sensitive to the test case. It gives the test case the flexibility to have its own sequence (should be different in the future).